dune_client/api/table.py

```diff
@@ -0,0 +1,115 @@
+"""
+Table API endpoints enables users to
+create and insert data into Dune.
+"""
+
+from __future__ import annotations
+from typing import List, Dict, Any
+
+from dune_client.api.base import BaseRouter
+from dune_client.models import DuneError
+
+
+class TableAPI(BaseRouter):
+    """
+    Implementation of Table endpoints - Plus subscription only
+    https://docs.dune.com/api-reference/tables/
+    """
+
+    def upload_csv(
+        self,
+        table_name: str,
+        data: str,
+        description: str = "",
+        is_private: bool = False,
+    ) -> bool:
+        """
+        https://docs.dune.com/api-reference/tables/endpoint/upload
+        The write API allows you to upload any .csv file into Dune. The only limitations are:
+
+        - File has to be < 200 MB
+        - Column names in the table can't start with a special character or digits.
+        - Private uploads require a Plus subscription.
+
+        Below are the specifics of how to work with the API.
+        """
+        response_json = self._post(
+            route="/table/upload/csv",
+            params={
+                "table_name": table_name,
+                "description": description,
+                "data": data,
+                "is_private": is_private,
+            },
+        )
+        try:
+            return bool(response_json["success"])
+        except KeyError as err:
+            raise DuneError(response_json, "UploadCsvResponse", err) from err
+
+    def create_table(
+        self,
+        namespace: str,
+        table_name: str,
+        schema: List[Dict[str, str]],
+        description: str = "",
+    ) -> Any:
+        """
+        https://docs.dune.com/api-reference/tables/endpoint/create
+        The create table endpoint allows you to create an empty table with a specifc schema in Dune.
+
+        The only limitations are:
+        - A table must currently be created as public.
+        - If the request tries to create a private table, it will fail.
+        - If a table already exists with the same name, the request will fail.
+        - Column names in the table can’t start with a special character or a digit.
+        """
+        payload = {
+            "schema": schema,
+            "description": description,
+        }
+        response_json = self._post(
+            route=f"/table/{namespace}/{table_name}/create", params=payload
+        )
+        try:
+            return response_json
```

Comment:
Is the convention here to return the json, or to define a data class with the response? Data class might be better? Currently I guess we don't return anything though!

Reply:
There seems to be only for `DuneError`.
I did follow what `upload_csv` did.  
We can modify this later, imo it's a minor detail. 

Lmk if you think otherwise

Reply:
Yeah, this shouldn't be blocking. But the `return response_json` can't fail, so you can remove the try except block here

Reply:
The idea is to return at least table name and namespace. We can wait to merge this until we have that change in place and we can create a proper return type for it.

Reply:
This is now done!